lib/storage/providers/IDBKeyVal.js

```diff
@@ -54,9 +55,17 @@ const provider = {
 
         return getValues.then((values) => {
             const upsertMany = _.map(pairs, ([key, value], index) => {
+                // If the value is null, we want to delete the key from storage,
+                // to be consistent with the native implementation with SQLite.
+                // SQLite by default removes keys from storage that are set to nullish values
+                if (_.isUndefined(value) || _.isNull(value)) {
```

Comment:
When would it be possible for `value` to be undefined? Wouldn't that mean that one of the `pairs` has only a key?

Reply:
The `value` part of the pair could also be `undefined` like `['something', undefined]`.

Reply:
I am sort of wondering how/why that happened though. It seems inconsistent with the merge behavior. Setting something to `null` has a special significance (i.e. delete me). Setting it to `undefined` does not mean anything. 

I would guess that we would just throw these values out vs. allow them to overwrite / delete any keys.

Reply:
Just added a `[HOLD on #334]` tag to this PR. The  [SQLite crash PR](https://github.com/Expensify/react-native-onyx/pull/334) actually fixes two things that are expected for this PR as well.

- Allow `undefined` values to be passed to `fastMerge`
- Remove key in `Onyx.merge` if `null` is passed

So we'll first want to merge the other PR in order for this to continue 👍

Reply:
This way, the extra condition in `IDBKeyVal.multiMerge` is not needed anymore